lib/auth/trustedcluster.go

```diff
@@ -67,8 +68,15 @@ func (a *Server) UpsertTrustedCluster(ctx context.Context, trustedCluster types.
 	}
 
 	// change state
+
+	if exists && !cmp.Equal(existingCluster.GetRoleMap(), trustedCluster.GetRoleMap()) {
+		if err := a.checkLocalRoles(ctx, trustedCluster.GetRoleMap()); err != nil {
+			return nil, trace.Wrap(err)
+		}
+		a.updateUserCARoleMap(ctx, existingCluster, trustedCluster)
+	}
 	switch {
-	case exists == true && enable == true:
+	case exists == true && enable == true && existingCluster.GetEnabled() != trustedCluster.GetEnabled():
```

Comment:
Can you store the result in a variable so we don't need to replace this check over and over?

```go
clusterDiffState := existingCluster.GetEnabled() != trustedCluster.GetEnabled()
```

I'm also opened to better naming.